src/docs/components/ComponentSettings.jsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const ComponentSettings = ({settings, values, onChange}) => {
+  function inputChanged(values, e) {
+    const target = e.target;
+    const key = e.target.dataset.key;
+    const type = e.target.dataset.valueType;
+    let value = null;
+
+    if (type === 'boolean') {
+      value = Boolean(target.checked);
+    } else if (type === 'number') {
+      value = Number(target.value);
+    } else if (type === 'object') {
+      value = JSON.parse(target.value);
+    } else if (type === 'array') {
+      value = values[target.value];
+    } else if (type === 'string') {
+      value = target.value;
+    }
+
+    onChange(key, value);
+  }
+
+  const content = Object.keys(settings).map(key => {
```

Comment:
Can we move it to separated component? And even 3 component, arrayComnponent, normalComponent